packages/babel-plugin-transform-typescript/src/namespace.js

```diff
@@ -0,0 +1,177 @@
+export default function transpileNamespace(path, t) {
+  if (path.node.declare || path.node.id.type === "StringLiteral") {
+    path.remove();
+    return;
+  }
+
+  const name = path.node.id.name;
+  const value = handleNested(path, t, t.cloneDeep(path.node));
+  const bound = path.scope.hasOwnBinding(name);
+  if (path.parent.type === "ExportNamedDeclaration") {
+    if (!bound) {
+      path.parentPath.insertAfter(value);
+      path.replaceWith(getDeclaration(t, name));
+      path.scope.registerDeclaration(path.parentPath);
+    } else {
+      path.parentPath.replaceWith(value);
```

Comment:
It's not actually valid typescript, but for some reason typescript wont fuss if the namespace itself is empty.

So, I'm a bit confused, what behavior should we be *testing*?

Reply:
I believe TypeScript ignores this because it doesn't emit anything for an empty namespace. That logic likely accidentally makes it ignore the type error as well.

Reply:
Oh ok; maybe typescript accepts it because it is considered as a type-only namespace?